src/AspNet.Security.OAuth.Apple/AppleAuthenticationExtensions.cs

```diff
@@ -76,15 +74,14 @@ public static AuthenticationBuilder AddApple([NotNull] this AuthenticationBuilde
             [CanBeNull] string caption,
             [NotNull] Action<AppleAuthenticationOptions> configuration)
         {
+            builder.Services.AddMemoryCache();
             builder.Services.TryAddSingleton<AppleClientSecretGenerator, DefaultAppleClientSecretGenerator>();
             builder.Services.TryAddSingleton<AppleIdTokenValidator, DefaultAppleIdTokenValidator>();
             builder.Services.TryAddSingleton<AppleKeyStore, DefaultAppleKeyStore>();
-            builder.Services.TryAddSingleton<JwtSecurityTokenHandler>();
 
             // Use a custom CryptoProviderFactory so that keys are not cached and then disposed of, see below:
             // https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1302
-            builder.Services.TryAddSingleton(
-                (_) => new CryptoProviderFactory() { CacheSignatureProviders = false });
+            builder.Services.AddSingleton<CustomCryptoProviderFactory>();
```

Comment:
I'd use an option for this one too. This way, you wouldn't need a custom derived type.

Reply:
I consider this an internal implementation detail to work around the caching behavior they changed which caused the bug it's working around in the first place, so I don't think we should expose it.

Reply:
Fine (that said, you can also have internal members in your options classes 😄)

Reply:
I did think of doing that, but then I thought it was a bit bleurgh. If you're happy with that, then I'll do that instead.

Reply:
This is still internal only, but now handled in `ApplePostConfigureOptions` rather than DI.